runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -146,9 +146,21 @@ public VersionLockedObject(Supplier<T> newObjectFn,
         if (ts != 0) {
             if (directAccessCheckFunction.apply(this)) {
                 log.trace("Access [{}] Direct (optimistic-read) access at {}", this, getVersionUnsafe());
-                R ret = accessFunction.apply(object);
-                if (lock.validate(ts)) {
-                    return ret;
+                try {
```

Comment:
I am not sure if we have seen a lot of performance benefits with optimistic locking here. Have we measured this against just simply taking a read lock ? The reason I ask is the subtle bugs and the code readability